go/oasis-test-runner/scenario/e2e/registry_cli.go

```diff
@@ -0,0 +1,699 @@
+		return fmt.Errorf("scenario/e2e/registry/entity: failed to submit entity register tx: %w", err)
+	}
+
+	// List entities.
+	entities, err = r.listEntities(childEnv)
+	if err != nil {
+		return err
+	}
+	// Three entities should now be registered after registration.
+	if len(entities) != 3 {
+		return fmt.Errorf("scenario/e2e/registry: initial entity list wrong number of entities: %d, expected at least: %d. Entities: %s", len(entities), 3, entities)
+	}
+
+	// Generate deregister entity transaction.
+	deregisterTxPath := filepath.Join(childEnv.Dir(), "registry_entity_deregister.json")
+	if err = r.genDeregisterEntityTx(childEnv, 1, deregisterTxPath, entDir.String()); err != nil {
+		return fmt.Errorf("scenario/e2e/registry/entity: failed to generate entity deregister tx: %w", err)
+	}
+
+	// Submit deregister entity transaction.
+	if err = r.submitTx(childEnv, deregisterTxPath); err != nil {
+		return fmt.Errorf("scenario/e2e/registry/entity: failed to submit entity deregister tx: %w", err)
+	}
+
+	// List entities.
+	entities, err = r.listEntities(childEnv)
+	if err != nil {
+		return err
+	}
+	// Only two entities should now be registered after deregistration.
+	if len(entities) != 2 {
+		return fmt.Errorf("scenario/e2e/registry: initial entity list wrong number of entities: %d, expected at least: %d. Entities: %s", len(entities), 2, entities)
+	}
+
+	return nil
+}
+
+// listEntities lists currently registered entities.
+func (r *registryCLIImpl) listEntities(childEnv *env.Env) ([]signature.PublicKey, error) {
+	r.logger.Info("listing all entities")
+	args := []string{
+		"registry", "entity", "list",
+		"--" + grpc.CfgAddress, "unix:" + r.basicImpl.net.Validators()[0].SocketPath(),
+	}
+	b, err := runSubCommandWithOutput(childEnv, "list", r.basicImpl.net.Config().NodeBinary, args)
+	if err != nil {
+		return nil, fmt.Errorf("scenario/e2e/registry/entity: failed to list entities: %s error: %w", b.String(), err)
+	}
+	entitiesStr := strings.Split(b.String(), "\n")
+
+	var entities []signature.PublicKey
+	for _, entStr := range entitiesStr {
+		// Ignore last newline.
+		if entStr == "" {
+			continue
+		}
+
+		var ent signature.PublicKey
+		if err = ent.UnmarshalHex(entStr); err != nil {
+			return nil, err
+		}
+		entities = append(entities, ent)
+	}
+
+	return entities, nil
+}
+
+// loadEntity loads entity and signer from given directory.
+func (r *registryCLIImpl) loadEntity(entDir string) (*entity.Entity, error) {
+	entitySignerFactory := fileSigner.NewFactory(entDir, signature.SignerEntity)
+	ent, signer, err := entity.Load(entDir, entitySignerFactory)
+	if err != nil {
+		return nil, fmt.Errorf("failed to load entity: %s", err)
+	}
+	if ent == nil {
+		return nil, fmt.Errorf("failed to load entity: no entity")
+	}
+	if signer.String() == "" {
+		return nil, fmt.Errorf("failed to load entity: empty signer")
+	}
+	if ent.ID.String() == "" {
+		return nil, fmt.Errorf("failed to load entity: empty entity")
+	}
```

Comment:
I don't think you need to test these as you are basically testing file signer for correctness here (and this should be done in the signer's unit test instead).